packages/dd-trace/src/appsec/rasp/index.js

```diff
@@ -86,7 +87,21 @@ function blockOnDatadogRaspAbortError ({ error }) {
 
   const { req, res, blockingAction } = abortError
   if (!isBlocked(res)) {
-    block(req, res, web.root(req), null, blockingAction)
+    const rootSpan = web.root(req)
+
+    try {
+      block(req, res, null, blockingAction)
+
+      rootSpan?.addTags({
+        'appsec.blocked': 'true'
+      })
+    } catch (err) {
+      rootSpan?.addTags({
+        '_dd.appsec.block.failed': 1
+      })
+
+      log.error('[ASM] Blocking error', err)
```

Comment:
this whole try catch is duplicated with the one in appsec/index.js isn't it ? why not just add this try catch directly in the block function ? so we don't have to change the code in those two places ?